apps/ledger-live-desktop/src/renderer/modals/SignMessage/utils.tsx

```diff
@@ -0,0 +1,48 @@
+import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
+import {
+  isEIP712Message,
+  getNanoDisplayedInfosFor712,
+} from "@ledgerhq/live-common/families/ethereum/hw-signMessage";
+import { MessageData } from "@ledgerhq/live-common/hw/signMessage/types";
+import { TypedMessageData } from "@ledgerhq/live-common/families/ethereum/types";
+import { NanoDisplayedInfoFor712 } from "./types";
+import { getEnv } from "@ledgerhq/live-common/env";
+
+export const getMessageProperties = async (
```

Comment:
This whole method is very horrible to look at.

Reply:
My code also has feelings you know...